Added tests for plugins

mspeedy opened the pull request:
Added tests for the database, storage, and cache plugins that mock the backend sources to test healthcheck functionality. All tests written as django TestCases, so the project needed to be a django project to run (as far as I could tell). Added necessary files to make it a runnable django project (manage.py, settings.py, etc.)

Commit:
Added tests for plugins and django modules to run them

Commit:
Added tests for plugins and django modules to run them

Commit:
Added tests for plugins and django modules to run them

Commit:
Added tests for plugins and django modules to run them

codingjoe commented on health_check/settings.py:
```diff
@@ -0,0 +1,129 @@
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.db.backends.sqlite3',
+        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
```
This should be `:memory:` which will result in the sql db being created in memory only.

codingjoe commented on health_check/wsgi.py:
```diff
@@ -0,0 +1,16 @@
+"""
```
You won't need a WSGI interface. I guess you can just drop the file

codingjoe commented:
Hi @mspeedy,

thanks for all that. I think it's a really good start. I have a couple of note, to integrate you tests a little smoother.
- py.test is able to run unittests. Thus there is absolutely no problem to mix them.
- As a bonus you don't need to setup a new app just to run the tests. You will need an app tho,
  if you want to test the integration. We have `tests/testapp/` already configured for that.
- Keep in mind the QA tests. You can execute the QA tests by calling `tox -e qa`

Let me know if you need any help. I can also fix everything while merging if that is ok with you.

-Joe

Commit:
Fixed styling so tox -e qa succeeds

Commit:
Fixed styling so tox -e qa succeeds

Commit:
Fixed bug in test_storage

Commit:
Fixed bug in test_storage

mspeedy commented:
Hi @codingjoe,
Thanks for your feedback! I'm still relatively new to django/python testing if you couldn't tell, so any help would be greatly appreciated. I've updated the project, removing the extra project files and making the QA tests pass. I'm now running the tests using py.test and all but the autodiscover test are passing, but I am unsure as to why. Let me know what other changes you think should be made.

Commit:
Added documentation for test and mock classes

Commit:
Added documentation for test and mock classes

aleksihakli committed:
Windows fixes (#50)

* Name tempfile in Windows compatible ISO8601

* Fix tempfile descriptors left open

codingjoe committed:
bump version number to 1.1.4

codingjoe commented:
@mspeedy ok, I'll try to look into it asap, but it might take a while since I'm about the leave for holiday

AstraLuma committed:
Add Migration

Add migration for `health_check_db`

codingjoe committed:
Bump version number to 1.1.5

AstraLuma committed:
Add released Django 1.10 to travis

joekiller committed:
include migrations for health_check_db in egg distribution.

Commit:
Bump version number

AstraLuma committed:
Add Django AppConfig integration (#55)

* Allow use of AppConfig

Actually, django has this algorithm in a util. Just use that.

* Minimize top-level imports

`__init__.py` is used by `setup.py`, so don't import anything at the top-level.

* pep8

* Use an AppConfig to do autodiscovery

Commit:
Bump version number to 1.2.0

aleksihakli committed:
Convert README from Markdown to reStructuredText

aleksihakli committed:
Add missing coding tags for UTF-8 compatibility

Close #65

codingjoe commented:
@mspeedy sorry It took me so long to get into this. Can you please rebase, I really want to merge this!

Commit:
Rebased and Fixed merge conflicts

Commit:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	health_check_db/migrations/0001_initial.py
#	health_check_db/plugin_health_check.py
#	health_check_storage/base.py
#	tests/test_backends.py

mspeedy commented:
Hey @codingjoe , This should fix things up. Worth noting though, I moved on to other projects and therefore never resolved the issue with test_autodiscover not passing. I believe the function still works properly, just not the test. You may want to look into it before merging just in case. 

Let me know if there's anything else you need!

aleksihakli commented:
Hey, seems like [Celery is missing](https://travis-ci.org/KristianOellegaard/django-health-check/jobs/183682633#L404) in tests, should that be added to [requirements-dev.txt](https://github.com/mspeedy/django-health-check/blob/master/requirements-dev.txt) @mspeedy?

codingjoe commented:
Merged in #69, thanks!
